modules/expo-background-notification-handler/ios/ExpoBackgroundNotificationHandlerModule.swift

```diff
@@ -0,0 +1,116 @@
+ * or killed
+ */
+public class ExpoBackgroundNotificationHandlerModule: Module {
+  let userDefaults = UserDefaults(suiteName: APP_GROUP)
+  
+  public func definition() -> ModuleDefinition {
+    Name("ExpoBackgroundNotificationHandler")
+    
+    OnCreate {
+      DEFAULTS.forEach { p in
+        if userDefaults?.value(forKey: p.key) == nil {
+          userDefaults?.setValue(p.value, forKey: p.key)
+        }
+      }
+    }
+    
+    AsyncFunction("getAllPrefsAsync") { () -> [String:Any]? in
+      var keys: [String] = []
+      DEFAULTS.forEach { p in
+        keys.append(p.key)
+      }
+      return userDefaults?.dictionaryWithValues(forKeys: keys)
+    }
+    
+    AsyncFunction("getBoolAsync") { (forKey: String) -> Bool in
+      if let pref = userDefaults?.bool(forKey: forKey) {
+        return pref
+      }
+      return false
+    }
+    
+    AsyncFunction("getStringAsync") { (forKey: String) -> String? in
+      if let pref = userDefaults?.string(forKey: forKey) {
+        return pref
+      }
+      return nil
+    }
+    
+    AsyncFunction("getStringArrayAsync") { (forKey: String) -> [String]? in
+      if let pref = userDefaults?.stringArray(forKey: forKey) {
+        return pref
+      }
+      return nil
+    }
+    
+    AsyncFunction("setBoolAsync") { (forKey: String, value: Bool) -> Void in
+      userDefaults?.setValue(value, forKey: forKey)
+    }
+    
+    AsyncFunction("setStringAsync") { (forKey: String, value: String) -> Void in
+      userDefaults?.setValue(value, forKey: forKey)
+    }
+    
+    AsyncFunction("setStringArrayAsync") { (forKey: String, value: [String]) -> Void in
+      userDefaults?.setValue(value, forKey: forKey)
+    }
+    
+    AsyncFunction("addToStringArrayAsync") { (forKey: String, string: String) in
+      if var curr = userDefaults?.stringArray(forKey: forKey),
+         !curr.contains(string)
+      {
+        curr.append(string)
+        userDefaults?.setValue(curr, forKey: forKey)
+      }
+    }
+    
+    AsyncFunction("removeFromStringArrayAsync") { (forKey: String, string: String) in
+      if var curr = userDefaults?.stringArray(forKey: forKey) {
+        curr.removeAll { s in
+          return s == string
+        }
+        userDefaults?.setValue(curr, forKey: forKey)
+      }
+    }
+    
+    AsyncFunction("addManyToStringArrayAsync") { (forKey: String, strings: [String]) in
+      if var curr = userDefaults?.stringArray(forKey: forKey) {
+        strings.forEach { s in
+          if !curr.contains(s) {
+            curr.append(s)
+          }
+        }
+        userDefaults?.setValue(curr, forKey: forKey)
+      }
+    }
+    
+    AsyncFunction("removeManyFromStringArrayAsync") { (forKey: String, strings: [String]) in
+      if var curr = userDefaults?.stringArray(forKey: forKey) {
+        strings.forEach { s in
+          curr.removeAll(where: { $0 == s })
+        }
+        userDefaults?.setValue(curr, forKey: forKey)
+      }
+    }
```

Comment:
We likely won't need all of these options for updating an array, but I just made some different ones I think we could run into. No harm no foul. (All we use right now is `setBool`, this would be for like mute words/muted threads/etc.